lib/elixir/lib/enum.ex

```diff
@@ -721,38 +721,49 @@ defmodule Enum do
   @spec fetch(t, index) :: {:ok, element} | :error
   def fetch(enumerable, index)
 
-  def fetch(first..last, index) when is_integer(index) do
-    fetch_range(first, last, index)
-  end
-
   def fetch(enumerable, index) when is_integer(index) and index < 0 do
-    case Enumerable.count(enumerable) do
-      {:ok, count} when (count + index) < 0 ->
-        :error
-
-      {:ok, count} ->
-        fetch_enumerable(enumerable, count + index, Enumerable)
-
-      {:error, _} ->
+    case enumerable do
```

Comment:
I think it's clearer to use function clauses rather than a `case`. The rest of the module uses function clauses to dispatch to different optimizations.